posthog/session_recordings/session_recording_api.py

```diff
@@ -643,7 +659,8 @@ def snapshots(self, request: request.Request, **kwargs):
         if personal_api_key:
             SNAPSHOTS_BY_PERSONAL_API_KEY_COUNTER.labels(api_key=personal_api_key, source=source).inc()
 
-        is_v2_enabled = posthoganalytics.feature_enabled(
+        # we let people opt in to the API via query param or enforce it via feature flag
+        is_v2_enabled = allow_blob_v2 or posthoganalytics.feature_enabled(
```

Comment:
question: How do the feature flags work with the API keys? Let's say we enable the feature flag for all teams, does it mean it will be enabled also for the people who use the API directly?

If that's the case, I think this condition has to be allow blob v2 AND feature flag enabled, plus we need to add the flag to the player.

Reply:
https://github.com/PostHog/posthog/issues/30734

Reply:
Ok, my concern was that I'd quite like to start gradually enabling blob v2 for more teams, so we might enable it for some people who are using the API directly. If the API required the `allow_blob_v2` param to serve v2 snapshots and we updated the player to pass it, we would be able to fully roll it out in the dashboard without affecting customers who use the API.

Reply:
Ah, I'm with you... if we make it so the API called is completely in control then at least their automation doesn't break and we buy time to identify and communicate

💖 love it... will update this PR

Reply:
updated and checked locally